src/model/duplicate.rs

```diff
@@ -0,0 +1,355 @@
+//! let des: Box<dyn DuplicateTraceConfig> = serde_json::from_str(config_file_content).unwrap();
+//! let mut model = des.into_model();
+//! assert_eq!(
+//!     model.next_duplicate(),
+//!     Some((vec![0.1, 0.2], Duration::from_secs(1)))
+//! );
+//! assert_eq!(
+//!     model.next_duplicate(),
+//!     Some((vec![0.2, 0.4], Duration::from_secs(1)))
+//! );
+//! assert_eq!(
+//!     model.next_duplicate(),
+//!     Some((vec![0.1, 0.2], Duration::from_secs(1)))
+//! );
+//! assert_eq!(
+//!     model.next_duplicate(),
+//!     Some((vec![0.2, 0.4], Duration::from_secs(1)))
+//! );
+//! assert_eq!(model.next_duplicate(), None);
+//! ```
+use crate::{DuplicatePattern, DuplicateTrace, Duration};
+use dyn_clone::DynClone;
+
+/// This trait is used to convert a duplicate trace configuration into a duplicate trace model.
+///
+/// Since trace model is often configured with files and often has inner states which
+///  is not suitable to be seialized/deserialized, this trait makes it possible to
+/// separate the configuration part into a simple struct for serialization/deserialization, and
+/// construct the model from the configuration.
+#[cfg_attr(feature = "serde", typetag::serde)]
+pub trait DuplicateTraceConfig: DynClone + Send {
+    fn into_model(self: Box<Self>) -> Box<dyn DuplicateTrace>;
+}
+
+dyn_clone::clone_trait_object!(DuplicateTraceConfig);
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+/// The model of a static duplicate trace.
+///
+/// ## Examples
+///
+/// ```
+/// # use netem_trace::model::StaticDuplicateConfig;
+/// # use netem_trace::{DuplicatePattern, Duration, DuplicateTrace};
+/// let mut static_duplicate = StaticDuplicateConfig::new()
+///     .duplicate(vec![0.1, 0.2])
+///     .duration(Duration::from_secs(1))
+///     .build();
+/// assert_eq!(static_duplicate.next_duplicate(), Some((vec![0.1, 0.2], Duration::from_secs(1))));
+/// assert_eq!(static_duplicate.next_duplicate(), None);
+/// ```
+#[derive(Debug, Clone)]
+pub struct StaticDuplicate {
+    pub duplicate: DuplicatePattern,
+    pub duration: Option<Duration>,
+}
+
+/// The configuration struct for [`StaticDuplicate`]
+///
+/// See [`StaticDuplicate`] for more details.
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize), serde(default))]
+#[derive(Debug, Clone, Default)]
+pub struct StaticDuplicateConfig {
+    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
+    pub duplicate: Option<DuplicatePattern>,
+    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
+    #[cfg_attr(
+        all(feature = "serde", feature = "human"),
+        serde(with = "humantime_serde")
+    )]
+    pub duration: Option<Duration>,
+}
+
+/// The model contains an array of duplicate trace models.
+///
+/// Combines multiple cuplicate trace models into one duplicate pattern,
```

Comment:
Wrong spelling: `cuplicate`? Also change `Combines` to `Combine`, since it is an imperative sentence (If there are also same mistakes in other files, correct them in a new PR)